File: docs/test-beds/cpp/fixture.cpp
// C++ test bed
#include <vector>
#include <string>
#include "helper.h"

#define MAX_ITEMS 100

template<typename T>
class Box {
public:
    T value;
    Box(T v) : value(v) {}
};

template<typename T>
class Container {
    std::vector<Box<T>> items;
public:
    void add(Box<T> item) { items.push_back(item); }
    int size() const { return items.size(); }
};

Container<std::string>* process(Box<std::string> input) {
    auto* c = new Container<std::string>();
    c->add(input);
    Helper::log(c);
    return c;
}

const int FOO = 42;

File: docs/test-beds/rust/fixture.rs
// Rust test bed
use std::collections::Vec;

pub struct Box<T> { pub value: T }

pub struct Container<T> { items: Vec<Box<T>> }

impl<T> Container<T> {
    pub fn new() -> Self { Container { items: Vec::new() } }
    pub fn add(&mut self, item: Box<T>) { self.items.push(item); }
    pub fn size(&self) -> usize { self.items.len() }
}

pub trait Logger {
    fn log(&self);
}

pub fn process(input: Box<String>) -> Container<String> {
    let mut c = Container::new();
    c.add(input);
    c
}

const FOO: i32 = 42;
